module/reset_domain/include/mod_reset_domain.h

```diff
@@ -0,0 +1,217 @@
+    MOD_RESET_DOMAIN_CAP_NOTIFICATION = (1UL << 31),
+    MOD_RESET_DOMAIN_CAP_ASYNC =  (1UL << 30)
+};
+
+/*!
+ * \brief Reset domain module configuration data.
+ *
+ */
+struct mod_reset_domain_config {
+    /*!
+     * \brief Notifications identifier.
+     */
+    fwk_id_t notification_id;
+
+};
+
+/*!
+ * \brief Reset domain element configuration data.
+ */
+struct mod_reset_domain_dev_config {
+    /*! Driver identifier */
+    fwk_id_t driver_id;
+
+    /*! Driver API identifier */
+    fwk_id_t driver_api_id;
+
+     /*! Supported modes (AUTORESET/AUTORESET ASYNC/EXPLICIT SIGNAL) */
+    uint32_t modes;
+
+    /*! Supported capabilities(ASYCN/NOTIFICATION SIGNAL) */
+    uint32_t capabilities;
+
+    /*! Supported reset states */
+    uint32_t reset_states;
+
+    /*! Reset latency */
+    unsigned int latency;
+};
+
+/*!
+ * \brief Reset domain HAL interface.
+ *
+ * \details The interface the reset domain clients relies on to perform
+ *      actions on a reset domain.
+ */
+struct mod_reset_domain_api {
+    /*!
+     * \brief Target reset domain \p dev_id performs a reset state change.
+     *
+     * \param dev_id Driver identifier of the reset domain.
+     *
+     * \param flags Bit mask of Signal Auto Reset or Auto Reset Async or
+           Explicit.
+     *
+     * \param reset_state  Bit mask of Reset Type and Reset ID.
+     *
+     * \param cookie Context-specific value.
+     *
+     * \retval FWK_SUCCESS or one of FWK_E_* error codes.
+     */
+    int (*set_reset_state)(fwk_id_t dev_id,
+                           uint32_t flags,
+                           uint32_t reset_state,
+                           uintptr_t cookie);
+
+};
+
+/*!
+ * \brief Reset domain driver interface.
+ *
+ * \details The interface this reset domain module relies on to perform
+ *      actions on a reset domain.
+ */
+struct mod_reset_domain_drv_api {
+    /*!
+     * \brief Target reset device \p dev_id performs an reset state change.
+     *
+     * \param dev_id Driver identifier of the reset domain.
+     *
+     * \param flags Bit mask of Signal Auto Reset or Auto Reset Async or
+           Explicit.
+     *
+     * \param reset_state  Bit mask of Reset Type and Reset ID.
+     *
+     * \param cookie Context-specific value.
+     *
+     * \retval FWK_SUCCESS or one of FWK_E_* error codes.
+     */
+    int (*set_reset_state)(fwk_id_t dev_id,
+                           uint32_t flags,
+                           uint32_t reset_state,
+                           uintptr_t cookie);
+
+};
+
+/*!
+ * \brief Reset domain notification indexes.
+ */
+enum mod_reset_domain_notify_idx {
+    MOD_RESET_DOMAIN_NOTIFICATION_AUTORESET,
+    MOD_RESET_DOMAIN_NOTIFICATION_IDX_COUNT
+};
+
+
+/*!
+ * \brief Reset domain auto reset notification event parameters.
+ */
+struct mod_reset_domain_notify_event_params {
+    /*! Domain identifier */
+    uint32_t domain_id;
+
+    /*! Reset state */
+    uint32_t reset_state;
```

Comment:
There's no documentation here on how this is represented so I can only presume it's the same as in the configuration structures, in which case this should probably also be `enum mod_reset_domain_reset_state`.

Reply:
done